CRM/Report/Form.php

```diff
@@ -1493,9 +1498,10 @@ public function buildInstanceAndButtons() {
 
     $this->assign('instanceForm', $this->_instanceForm);
 
-    // CRM-16274 Determine if user has 'edit all contacts' or equivalent
-    $permission = CRM_Core_Permission::getPermission();
-    if ($permission == CRM_Core_Permission::EDIT &&
+    // CRM-19330 - check that the user has Edit permissions
+    // getPermission() simply returns CRM_Core_Permission::EDIT.
+    $permission = CRM_Core_Permission::check(CRM_Core_Permission::EDIT);
+    if ($permission  &&
```

Comment:
With this change, a user with `edit all contacts` permission is unable to view `add2group` dropdown. I think this should be visible as with this permission user can add contact to group from search result, contact summary page etc.